### PR TITLE
refactor: switched from single to multi selector for projects categories

### PR DESCRIPTION
**Pull request by 0xjei**
## Description

This PR enable the selection of multiple project categories. The projects shown are those that have at least all categories selected.

## Related Issue
See #459 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

## Other information

[Screencast from 12-12-2023 00:49:15.webm](https://github.com/semaphore-protocol/semaphore/assets/20580910/18b7bb7e-62f6-4157-b2ab-202f9b431d72)

### CODE REVIEW

File: apps/website/src/components/ProjectsList.tsx [hunk 1]
```diff
@@ -12,32 +12,18 @@ import { chunkArray } from "../utils/chunkArray"
...
 
 export default function ProjectsList(props: any) {
-    const [projects, setProjects] = useState<(typeof allProjects)[]>(chunkArray(allProjects))
+    const [projects] = useState<(typeof allProjects)[]>(chunkArray(allProjects))
     const [index, setIndex] = useState<number>(0)
-    const [selectedCategory, setSelectedCategory] = useState<string | null>(null)
     const [onlyPSE, setOnlyPSE] = useState<boolean | null>(null)
+    const [selectedCategories, setSelectedCategories] = useState<Array<string>>([])
 
-    const filterProjects = useCallback(() => {
```
Comment (cedoor) [inline apps/website/src/components/ProjectsList.tsx:20]: I think it's better to keep this function here and make it work with multi-categories. There were two different filters: 1 for PSE projects and 1 for tags and they should work together (that's why the logic is a little bit more complex than 1 filter). With your function, you're only filtering by tags/categories (nothing happens if I click on PSE/Community).

**Review by cedoor — CHANGES_REQUESTED** [review]

File: apps/website/src/components/ProjectsList.tsx (hunk 1 above)
Comment (0xjei) [reply apps/website/src/components/ProjectsList.tsx:20]: You right, fixed in a6fde2fdedb3ad70970e643b4a212364ea77c6fa

File: apps/website/src/components/ProjectsList.tsx [hunk 2]
```diff
@@ -66,23 +68,25 @@ export default function ProjectsList(props: any) {
                 </HStack>
             </VStack>
 
-            <VStack align="left" spacing="6">
-                <Text fontSize="20">Category</Text>
```
Comment (cedoor) [inline apps/website/src/components/ProjectsList.tsx:70]: I think we should keep this text.

**Review by cedoor — CHANGES_REQUESTED** [review]

File: apps/website/src/components/ProjectsList.tsx (hunk 2 above)
Comment (0xjei) [reply apps/website/src/components/ProjectsList.tsx:70]: I must have missed it, my bad!

**Review by cedoor — APPROVED** [review]
LGTM!

**Review by vplasencia — APPROVED** [review]

**Pull request merged**